CI is failing because of updated `unist` types

Due to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54413, CI is now failing with

```sh
Error: [typecheck] src/index.ts(90,14): error TS2339: Property 'title' does not exist on type 'Literal<unknown, Data> & Parent<Node<Data>, Data>'.
Error: [typecheck] src/index.ts(92,26): error TS2339: Property 'value' does not exist on type 'Node<Data>'.
Error: [typecheck] src/index.ts(92,41): error TS2339: Property 'url' does not exist on type 'Literal<unknown, Data> & Parent<Node<Data>, Data>'.
Error: [typecheck] src/index.ts(97,14): error TS2339: Property 'url' does not exist on type 'Literal<unknown, Data> & Parent<Node<Data>, Data>'.
Error: [typecheck] src/index.ts(163,32): error TS2339: Property 'tagName' does not exist on type 'Parent<Node<Data>, Data>'.
Error: [typecheck] src/index.ts(164,38): error TS2339: Property 'properties' does not exist on type 'Parent<Node<Data>, Data>'.
```

@wooorm Can you point me into the right direction of fixing this please?
I don't have a clue tbh. 🤔

--- a/README.md
+++ b/README.md
@@ -173,14 +173,16 @@ twitter for example][twitter-oembed-docs]).
 
 #### `handleHTML?: (html: GottenHTML, info: TransformerInfo) => GottenHTML | Promise<GottenHTML>`
 
-Add optional HTML around what is returned by the transformer. This is useful
-for surrounding the returned HTML with custom HTML and classes.
+Add optional HTML around what is returned by the transformer. This is useful for
+surrounding the returned HTML with custom HTML and classes.
 
-Here's a quick example of an HTML handler that would handle adding [TailwindCSS aspect ratio](https://github.com/tailwindlabs/tailwindcss-aspect-ratio) classes to YouTube videos:
+Here's a quick example of an HTML handler that would handle adding
+[TailwindCSS aspect ratio](https://github.com/tailwindlabs/tailwindcss-aspect-ratio)
+classes to YouTube videos:
 
 ```typescript
 import remark from 'remark'
-import remarkEmbedder, { TransformerInfo } from '@remark-embedder/core'
+import remarkEmbedder, {TransformerInfo} from '@remark-embedder/core'
 import oembedTransformer from '@remark-embedder/transformer-oembed'
 import remarkHtml from 'remark-html'
 
@@ -191,10 +193,10 @@ https://www.youtube.com/watch?v=dQw4w9WgXcQ
 `
 
 function handleHTML(html: string, info: TransformerInfo) {
-  const { url, transformer } = info
+  const {url, transformer} = info
   if (
-    transformer.name === '@remark-embedder/transformer-oembed'
-    || url.includes('youtube.com')
+    transformer.name === '@remark-embedder/transformer-oembed' ||
+    url.includes('youtube.com')
   ) {
     return `<div class="embed-youtube aspect-w-16 aspect-h-9">${html}</div>`
   }
@@ -209,9 +211,9 @@ const result = await remark()
   .use(remarkHtml)
   .process(exampleMarkdown)
 
-  // This should return:
-  // <p>Check out this video:</p>
-  // <div class="embedded-youtube aspect-w-16 aspect-h-9"><iframe width="200" height="113" src="https://www.youtube.com/embed/dQw4w9WgXcQ?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
+// This should return:
+// <p>Check out this video:</p>
+// <div class="embedded-youtube aspect-w-16 aspect-h-9"><iframe width="200" height="113" src="https://www.youtube.com/embed/dQw4w9WgXcQ?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
 ```
 
 #### `handleError?: (errorInfo: ErrorInfo) => GottenHTML | Promise<GottenHTML>`

--- a/package.json
+++ b/package.json
@@ -38,19 +38,19 @@
     "validate": "kcd-scripts validate"
   },
   "dependencies": {
-    "@babel/runtime": "^7.13.10",
-    "@types/unist": "^2.0.3",
+    "@babel/runtime": "^7.14.6",
+    "@types/mdast": "^3.0.6",
     "hast-util-from-parse5": "^6.0.1",
     "parse5": "^6.0.1",
-    "unified": "^9.2.1",
+    "unified": "^9.2.2",
     "unist-util-visit": "^2.0.3"
   },
   "devDependencies": {
-    "@types/jest": "^26.0.22",
-    "kcd-scripts": "^9.0.0",
+    "@types/jest": "^26.0.24",
+    "kcd-scripts": "^11.1.0",
     "remark": "^13.0.0",
     "remark-html": "^13.0.1",
-    "typescript": "^4.2.4"
+    "typescript": "^4.3.5"
   },
   "eslintConfig": {
     "extends": "./node_modules/kcd-scripts/eslint.js"

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,7 +1,7 @@
 import fromParse5 from 'hast-util-from-parse5'
 import parse5 from 'parse5'
 import type {Plugin} from 'unified'
-import type {Literal, Parent} from 'unist'
+import type {Link, Parent, Text} from 'mdast'
 import visit from 'unist-util-visit'
 
 type GottenHTML = string | null
@@ -82,19 +82,19 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
       }
 
       const {children} = paragraphNode
-      const node = children[0] as Literal & Parent
+      const node = children[0] as Link | Text
       const isText = node.type === 'text'
       // it's a valid link if there's no title, and the value is the same as the URL
       const isValidLink =
         node.type === 'link' &&
-        node.title === null &&
+        !node.title &&
         node.children.length === 1 &&
         node.children[0].value === node.url
       if (!isText && !isValidLink) {
         return
       }
 
-      const {url, value = url} = node
+      const {url, value = url} = node as Link
 
       const urlString = getUrlString(value as string)
       if (!urlString) {
@@ -104,7 +104,7 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
     })
 
     const nodesToTransform: Array<
-      typeof nodeAndURL[0] & typeof transformersAndConfig[0]
+      typeof nodeAndURL[number] & typeof transformersAndConfig[number]
     > = []
 
     for (const node of nodeAndURL) {

